doc/arch/007-screen-architecture-v2.md

```diff
@@ -0,0 +1,80 @@
+# ADR 007: Screen Architecture (v2)
+
+## Status
+Accepted (Supersedes [001](001-screen-controllers.md)) on 2019-07-03
+
+## Context
+Our [current screen architecture](001-screen-controllers.md) has a bunch of problems:
+
+- We currently recreate application state from the UI state, which is dependent on some hidden behaviour
+	- RxBinding has initial value observables that emit the current event immediately.
+	- All widgets on the screen save and restore state properly.
+- The entire event stream is replayed because the system is setup in a way that they begin emission of events like ScreenCreated before the entire event handling loop is setup.
+- Controller handles both the business logic and the view logic.
+
+All of these things combine together to make testing and maintaining the screens hard. In addition, screens which perform a lot of business logic in memory require a lot of working around the fact that the architecture depends on the state being saved either in persistence or by the platform mechanisms.
+
+## Decision
+
+### Goals
+- Separate presentation and business logic so that they can tested independently of each other.
+- Make UI state explicit and save/restore it manually instead of depending on hidden behaviour.
+- Should be easy to migrate from the older architecture to the new for existing screens.
+
+### Implementation
+We need to split the controller into two discrete pieces, each with their own responsibility.
+
+- `UiStateProducer`: This will be responsible for performing the function of the controller related to the business logic.
```

Comment:
Maybe add a reference to examples of `UiStateProducer` and `UiChangeProducer` referring to a specific commit.

Reply:
Good point, I have done this.